presto-spi/src/main/java/io/prestosql/spi/connector/CatalogSchemaProcedureName.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spi.connector;
+
+import java.util.Objects;
+
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+public final class CatalogSchemaProcedureName
```

Comment:
We should probably call this `CatalogSchemaObjectName` so that it's more generic. That way we can reuse it for functions. Same with `SchemaObjectName`.

Reply:
Actually, I'm not sure if there are any other schema-scoped objects that would matter, so perhaps naming this `CatalogSchemaRoutineName` would suffice.